### DISCUSSION TITLE
The Road to 1.0

### DISCUSSION THREAD

**Discussion by leoafarias** [btwld/mix#90] [General]
There has been a lot of progress made in the past few months, and I am thrilled with the initial API decisions and the approach to its composability. We will break down some focus areas in this discussion but also allow anyone to make suggestions on anything that might be missing for a sable 1.0 release.

## Documentation
Mix is a relatively simple implementation for the visual properties, but some concepts require explanation, and there need to be more examples and best practices information. We will be focusing more on making Mix easier to onboard.

- Concepts
- Automated API Reference (dartdoc)
- Best practices
- Getting started example
- Sample code and recipes

## Performance
There has been an initial pass on Mix to ensure it is performant. We are monitoring some Flutter/Dart initiatives that will further help with this. But this outlines a small roadmap to guarantee everyone using Mix can be confident of its high performance.

- Create automated performance benchmarks 
- Review inherited widget Mix caching
- Explore macros/build values to remove any overhead
- Create best practice documentation and information for best performance

## Headless Widgets
We have been playing with the concept of widgets that people can use as the foundation for their design systems. I believe we have defined good best practices and have created some initial widgets, and we are pleased with the results.

However, this is an ongoing initiative, and we are looking for feedback on what should be part of headless widgets or not.

- Finish initial "Atoms" of components
- Explore form component implementations (TextField, Toggle, RadioGroups)
- Public release an entire design system using headless widgets to highlight best practices and approaches.

## Bugs, Enhancements, and Stability
We were approached by some companies interested in Mix and using it to build their design systems in Flutter. We plan on allocating our attention and resources to support any company planning to use Mix in production; this will be the highest priority. We believe the feedback and improvements from these conversations are invaluable.

We are very excited and look forward to showing what we have been working on in the next few weeks.

**Comment by nobrefelipe** [comment]
Can't wait to be able to use it in production here at Shift, we are about to develop our UI Kit and Mix would be a great tool for it.

**Reply by leoafarias** [reply]
@nobrefelipe would love to hear more about your specific case, and understand some of your current pain points and what you expect Mix to solve.
